README.md

```diff
@@ -134,7 +128,7 @@ if err != nil {
 }
 
 // Use PageIterator to iterate through all users
-pageIterator, err := msgraphcore.NewPageIterator(result, adapter, models.CreateUserCollectionResponseFromDiscriminatorValue)
+pageIterator, err := msgraphcore.NewPageIterator(result, client.GetAdapter(), models.CreateUserCollectionResponseFromDiscriminatorValue)
```

Comment:
If we defined an interface in core (which I think we should do) that contains the getter, we could simply ask for the client here and make the type assertion in the iterator, simplifying things

Alternatively we could hold on the change I'm suggesting until we have a decision on #257

Reply:
I think we can have the discussion on this separately, I do have a few syntax styles on the same we can weigh on

Reply:
fair enough, resolving this one.